### DISCUSSION TITLE
.sourceCodeLocation

### DISCUSSION THREAD

**Discussion by danielbayley80** [cheeriojs/cheerio#3020] [Q&A]
Hi folks.

I've been using Cheerio for some time and it works great. I upgraded recently and got it working but can no longer figure out how to get the node.sourceCodeLocation option to work.

previously I would use the CheerioAPI and then on load add the option object with the .sourceCodeLocation property.

then I would reference the property as node.sourceCodeLocation.

I can no longer see a mechanism to do this, nor can I find a working example for the new version.

Can you give me some pointers please, assuming this is still possible?

**Comment by fb55** [comment]
I don't think anything has changed here. We have a test for this that is passing, so it should still work:

https://github.com/cheeriojs/cheerio/blob/b3bc6c8e002740d30bec99906da4af45005867ee/src/cheerio.spec.ts#L443-L449

**Comment by danielbayley80** [comment]
Hmmm Perhaps something changed with the type definitions / interfaces? 
I'm using strongly typed typescript, maybe the property works but is not defined?
I'll dig a little more.

**Comment by danielbayley80** [comment]
![image](https://user-images.githubusercontent.com/1572154/221005124-6e3176b4-6d53-4adc-9384-4655dca0907d.png)

![image](https://user-images.githubusercontent.com/1572154/221005362-662b6328-cffb-4955-ba1a-7aa8acda40fe.png)

**Reply by fb55** [reply]
Looks like you'll have to remove `@types/cheerio` and use the types shipped with the package.

**Reply by danielbayley80** [reply]
Ah, okay, thanks, I hadn't spotted that they were out of sync.. that fixed everything. thanks.
